### PR TITLE
docs(nav): nav link component playground

### PR DESCRIPTION
**Pull request by sean-perkins**
Quick link: https://ionic-docs-git-docs-nav-link-playground-ionic1.vercel.app/docs/api/nav#using-navlink

To verify the React example, you will need to manually install the dev-build: `6.1.12-dev.11656705976.18fe5a46`

### CODE REVIEW

File: docs/api/nav.md [hunk 1]
```diff
@@ -42,7 +42,11 @@ TODO: Playground Example
 
 ## Using NavLink
 
-TODO: Playground Example
+NavLink is a simplified API when interacting with Nav. Developers can customize the component, pass along component properties, modify the direction of the route animation or define a custom animation when navigating. 
+
+import NavLinkExample from '@site/static/usage/nav/nav-link/index.md';
+
+<NavLinkExample />
```
Comment (amandaejohnston) [inline docs/api/nav.md:49]: Would it be possible to get rid of the extra space in and above the header in the inline demo?
ios
md

File: static/usage/nav/nav-link/vue/page_three_vue.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+```html
+<template>
+  <ion-page>
+    <ion-header>
+      <ion-toolbar>
+        <ion-buttons slot="start">
+          <ion-back-button></ion-back-button>
```
Comment (amandaejohnston) [inline static/usage/nav/nav-link/vue/page_three_vue.md:7]: This back button doesn't seem to be showing in the Vue Stackblitz.

File: static/usage/nav/nav-link/vue/page_two_vue.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+```html
+<template>
+  <ion-page>
+    <ion-header>
+      <ion-toolbar>
+        <ion-title>Page Two</ion-title>
```
Comment (amandaejohnston) [inline static/usage/nav/nav-link/vue/page_two_vue.md:6]: This page is missing the `ion-back-button`.

File: static/usage/nav/nav-link/angular/page_one_component_ts.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  template: `
+    <ion-header>
+      <ion-toolbar>
+        <ion-title>Page One</ion-title>
+      </ion-toolbar>
+    </ion-header>
+    <ion-content class="ion-padding">
+      <h1>Page One</h1>
+      <ion-nav-link router-direction="forward" [component]="component">
+        <ion-button>Go to Page Two</ion-button>
```
Comment (amandaejohnston) [inline static/usage/nav/nav-link/angular/page_one_component_ts.md:17]: This button isn't doing anything when clicked in the Angular Stackblitz, possibly the other buttons too.

File: static/usage/nav/nav-link/react/page_two_tsx.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+import PageThree from './page-three';
+
+function PageTwo() {
+  return (
+    <IonPage>
+      <IonHeader>
+        <IonToolbar>
+          <IonButtons slot="start">
+            <IonBackButton></IonBackButton>
```
Comment (amandaejohnston) [inline static/usage/nav/nav-link/react/page_two_tsx.md:23]: The back buttons aren't showing up in the React Stackblitz.

**Review by amandaejohnston — CHANGES_REQUESTED** [review]

File: docs/api/nav.md (hunk 1 above)
Comment (liamdebeasi) [reply docs/api/nav.md:49]: ^ Maybe we just have this demo use the phone shell? Might look a bit nicer

File: static/usage/nav/nav-link/angular/page_one_component_ts.md (hunk 4 above)
Comment (liamdebeasi) [reply static/usage/nav/nav-link/angular/page_one_component_ts.md:17]: Looks like a circular dependency?

```
Unhandled Promise rejection:
NG0200: Circular dependency in DI detected for NavController. Find more at https://angular.io/errors/NG0200
```

File: static/usage/nav/nav-link/vue/page_three_vue.md (hunk 2 above)
Comment (liamdebeasi) [reply static/usage/nav/nav-link/vue/page_three_vue.md:7]: I think the problem is that the components use `<ion-page>`. That component should only be used with the router.

There's probably an improvement we can make here where developers need to reuse components in routing and non-routing contexts.

Comment (liamdebeasi) [reply static/usage/nav/nav-link/vue/page_three_vue.md:7]: Ah actually I think we already got an issue like this: https://github.com/ionic-team/ionic-framework/issues/23304

**Review by liamdebeasi — APPROVED** [review]
Same comments as Amanda. Otherwise looks great. Nice job!

**Review by amandaejohnston — APPROVED** [review]
Looking good now!

**Pull request merged**